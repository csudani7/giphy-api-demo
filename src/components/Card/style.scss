.cardContent {
    position: relative;
    font-size: 30px;
    text-align: center;
    margin: 17px;
    box-shadow: 11px 11px 4px #a1a1a3;
    border-radius: 11px;
}

.cardContent:hover {
    transform: scale(1.3);
    box-shadow: none;
}

.delete-gif-icon {
    left: 83%;
    top: 7%;
    cursor: pointer;
    position: absolute;
    color: white;
    background: #3d3d3d;
    padding: 3px;
    border-radius: 30rem;
}
