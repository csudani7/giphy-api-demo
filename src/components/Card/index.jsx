import React from 'react';
import Card from '@mui/material/Card';
import DeleteIcon from '@mui/icons-material/Delete';

import './style.scss';

export default function GifCard(props) {
    const { imageUrl, deleteSelectedGif, selectedId } = props;

    return (
        <div className="cardContent" style={{ border: `6px solid rgb(${Math.floor(Math.random() * 111)}, ${Math.floor(Math.random() * 111)}, ${Math.floor(Math.random() * 111)})` }}>
            <DeleteIcon onClick={() => deleteSelectedGif(selectedId)} className='delete-gif-icon' />
            <Card sx={{ maxWidth: 345 }}>
                <iframe src={imageUrl} title={imageUrl} width="200" height="200" frameBorder="0" className="giphy-embed" />
            </Card>
        </div>

    );
}
