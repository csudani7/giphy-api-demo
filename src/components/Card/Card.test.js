import React from "react";
import { render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "react-query";

import Card from ".";

describe("Card", () => {
  it("should render Card page error free", () => {
    const queryClient = new QueryClient();
    function TestCard() {
      return (
        <QueryClientProvider client={queryClient}>
          <Card />
        </QueryClientProvider>
      );
    }
    const { asFragment } = render(<TestCard />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
