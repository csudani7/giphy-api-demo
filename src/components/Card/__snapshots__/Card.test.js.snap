// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render Card page error free 1`] = `
<DocumentFragment>
  <div
    class="cardContent"
    style="border: 6px solid rgb(71, 50, 58);"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium delete-gif-icon css-i4bv87-MuiSvgIcon-root"
      data-testid="DeleteIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
      />
    </svg>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
    >
      <iframe
        class="giphy-embed"
        frameborder="0"
        height="200"
        width="200"
      />
    </div>
  </div>
</DocumentFragment>
`;
